cmd/podman/common/create.go

```diff
@@ -396,8 +396,8 @@ func GetCreateFlags(cf *ContainerCLIOpts) *pflag.FlagSet {
 	)
 	createFlags.StringVar(
 		&cf.ShmSize,
-		"shm-size", containerConfig.ShmSize(),
-		"Size of /dev/shm "+sizeWithUnitFormat,
+		"shm-size", "",
```

Comment:
Why did you take away default?

Reply:
It's meant to know in the backend if it has been specified or not. A comment would help to make it explicit - some might want to make it consistent in the future.

Reply:
I would prefer to move away from using the create config struct and actually checking to see if the user specified options via cobra, and then telling the user what the true default is via cobra, rather then hacking each one up.

On the Remote case we can show them as "".

Reply:
can we address this separately?  Otherwise it currently fails on each container that has a IPC namespace with:

```
Error: invalid config provided: cannot set shmsize when creating an IPC namespace
```

Reply:
Why can't it?  I think when creating a IPC Namespace this should be fine, when sharing an IPC Namespace it should not be allowed.

Reply:
the error message is wrong, but the check is correct:

```
	if s.ContainerStorageConfig.ShmSize != nil && !s.ContainerStorageConfig.IpcNS.IsPrivate() {
		return errors.New("cannot set shmsize when creating an IPC namespace")
	}
```

we cannot set the shmsize if the IPC is not private.

I got confused by the error message, I get the error when using `--ipc=host`.

I've added another patch to fix the error message

Reply:
Can't we check if the flag is set and only add it to SpecGen if that is true?